octue/resources/manifest.py

```diff
@@ -13,34 +13,48 @@
 class Manifest(Pathable, Serialisable, Identifiable, Hashable):
     """A representation of a manifest, which can contain multiple datasets This is used to manage all files coming into
     (or leaving), a data service for an analysis at the configuration, input or output stage.
+
+    :param str|None id: the UUID of the manifest (a UUID is generated if one isn't given)
+    :param str|None path: the path the manifest exists at (defaults to the current working directory)
+    :param dict|None datasets: a mapping of dataset names to `Dataset` instances, serialised datasets, or paths to datasets
+    :return None:
     """
 
     _ATTRIBUTES_TO_HASH = ("datasets",)
     _SERIALISE_FIELDS = "datasets", "keys", "id", "name", "path"
 
-    def __init__(self, id=None, path=None, datasets=None, keys=None, **kwargs):
+    def __init__(self, id=None, path=None, datasets=None, **kwargs):
```

Comment:
I don't feel right about having this here in the init method. We'll talk more about why that is in the coming months (work with the IEA will draw out the need to translate and provide transformers between schema versions).

Can you actually pull it right out into a function, perhaps in a 'migrations" module?

That makes it very clear what it's for - keeping the piece of code we have to deprecate out of the main program logic but also providing me a reference point that I can think about establishing a more formalised data migration process on top of.

Reply:
Yep will do. Will it be removed from the `migrations` module in the next breaking change release?

Reply:
Also did you want the same thing done in e.g. `twined` next time?